lib/Runtime/ByteCode/ByteCodeSerializer.cpp

```diff
@@ -2233,8 +2233,9 @@ class ByteCodeBufferBuilder
                   | FunctionInfo::Attributes::CapturesThis
                   | FunctionInfo::Attributes::Generator
                   | FunctionInfo::Attributes::ClassConstructor
-                  | FunctionInfo::Attributes::ClassMethod)) == 0,
-                "Only the ErrorOnNew|SuperReference|Lambda|CapturesThis|Generator|ClassConstructor|Async|ClassMember attributes should be set on a serialized function");
+                  | FunctionInfo::Attributes::ClassMethod
+                  | FunctionInfo::Attributes::EnclosedByGlobalFunc)) == 0,
```

Comment:
I would really like for there to be some static FunctionInfo::SerializableFunctionAttributes FunctionInfo::NonSerializableFunctionAttributes that would break at compile time when we add a new attribute, but that's out of the scope of this.